### PR TITLE
ticdc: add detailed instructions for verifying if TiCDC has synchronized all updates to downstream

### PR DESCRIPTION
**Pull request by wlwilliamx**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

#### Overview
This PR introduces a comprehensive guide on how to check if TiCDC has synchronized all updates from an upstream TiDB cluster to downstream systems. These steps ensure that users can confidently verify the synchronization completeness of TiCDC in scenarios where upstream updates have stopped.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.5 (TiDB 8.5 versions)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]

File: ticdc/ticdc-faq.md [hunk 1]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
 >
 > 该功能在 TiCDC 4.0.3 版本引入。
 
+## 上游停止更新后，如何判断 TiCDC 是否已将所有更新同步到下游？
+
+可以通过对比时间戳的方式查看。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:58]:
```suggestion
在上游 TiDB 集群停止更新后，可以通过比较上游 TiDB 集群的最新 [TSO](/glossary.md#tso) 时间戳与 TiCDC 当前的同步进度判断同步是否完成。如果 TiCDC 的同步进度时间大于或等于上游 TiDB 集群的 TSO，则说明 TiCDC 已同步所有更新。具体操作步骤如下：
```

File: ticdc/ticdc-faq.md [hunk 2]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+1. 在上游 TiDB 停止所有更新后，获取上游 TiDB 集群的最新 TSO 对应的时间。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:60]:
```suggestion
1. 获取上游 TiDB 集群的最新 TSO 时间戳。
```

File: ticdc/ticdc-faq.md [hunk 3]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+    > 注意：
+    >
+    > 不要直接使用 `NOW()` 等查看时间的方式，要用 TSO，这里是为了方便后面的时间对比所以将 TSO 转成时间格式。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:64]:
```suggestion
    > **注意：**
    >
    > 请使用 [`TIDB_CURRENT_TSO()`](/functions-and-operators/tidb-functions.md#tidb_current_tso) 函数获取 TSO，而不是 `NOW()` 等查询当前时间的函数。
```

File: ticdc/ticdc-faq.md [hunk 4]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:65]:
```suggestion

    以下示例使用 [`TIDB_PARSE_TSO()`](/functions-and-operators/tidb-functions.md#tidb_parse_tso) 将 TSO 转换为可读的时间格式，便于后续比较：

```

File: ticdc/ticdc-faq.md [hunk 5]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+    ```sql
+    BEGIN; SELECT TIDB_PARSE_TSO(TIDB_CURRENT_TSO()); ROLLBACK;
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:67]:
```suggestion
    BEGIN;
    SELECT TIDB_PARSE_TSO(TIDB_CURRENT_TSO());
    ROLLBACK;
```

File: ticdc/ticdc-faq.md [hunk 6]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+    ```
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:69]:
```suggestion

    输出结果示例如下：

```

File: ticdc/ticdc-faq.md [hunk 7]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+    ```sql
+    +------------------------------------+
+    | TIDB_PARSE_TSO(TIDB_CURRENT_TSO()) |
+    +------------------------------------+
+    | 2024-11-12 20:35:34.848000         |
+    +------------------------------------+
+    ```
+
+2. 获取 TiCDC 当前的同步进度，如果进度大于等于刚刚获得的上游 TSO，就表示已经同步了所有更新。获取当前同步进度有两种方式：
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:78]:
```suggestion
2. 获取 TiCDC 当前的同步进度。

    你可以通过以下两种方法之一检查当前的同步进度：
```

File: ticdc/ticdc-faq.md [hunk 8]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+    * **方法一**：通过 `cdc` 查看所有同步任务的 Checkpoint。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:80]:
```suggestion
    * **方法一**：查询同步任务的 Checkpoint（推荐）
```

File: ticdc/ticdc-faq.md [hunk 9]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:81]:
```suggestion

        使用 [TiCDC 命令行工具](/ticdc/ticdc-manage-changefeed.md) `cdc cli` 查看所有同步任务的 Checkpoint：

```

File: ticdc/ticdc-faq.md [hunk 10]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+      ```shell
+      cdc cli changefeed list --server=http://127.0.0.1:8300
+      ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:84]:
```suggestion
        ```shell
        cdc cli changefeed list --server=http://127.0.0.1:8300
        ```
```

File: ticdc/ticdc-faq.md [hunk 11]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+        在输出中可以看到所有同步任务的 `checkpoint` 字段：
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:86]:
```suggestion
        输出结果示例如下：
```

File: ticdc/ticdc-faq.md [hunk 12]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+          "namespace": "default",
+          "summary": {
+            "state": "normal",
+            "tso": 453880043653562372,
+            "checkpoint": "2024-11-12 20:36:01.447",
+            "error": null
+          }
+        }
+      ]
+      ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:101]:
```suggestion
        ```json
        [
          {
            "id": "syncpoint",
            "namespace": "default",
            "summary": {
              "state": "normal",
              "tso": 453880043653562372,
              "checkpoint": "2024-11-12 20:36:01.447",
              "error": null
            }
          }
        ]
        ```
```

File: ticdc/ticdc-faq.md [hunk 13]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      `"checkpoint": "2024-11-12 20:36:01.447"` 表示 TiCDC 当前的同步进度已达到 `2024-11-12 20:36:01.447`，上游 TiDB 集群在这个时间之前的所有变更已同步完。所以当 checkpoint 大于等于刚刚获取的上游 TiDB 的 TSO 时间就说明所有更新已同步到下游。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:103]:
```suggestion
        在输出结果中，`"checkpoint": "2024-11-12 20:36:01.447"` 表示 TiCDC 当前的同步进度为 `2024-11-12 20:36:01.447`，即所有在该时间之前的上游 TiDB 变更已同步完成。如果该时间戳大于或等于步骤 1 中获取的上游 TiDB TSO 时间，则表示所有更新已经同步到下游。
```

File: ticdc/ticdc-faq.md [hunk 14]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+    * **方法二**：若下游为 TiDB，且开启了 TiCDC Syncpoint 功能，那么可以通过查看 Syncpoint 信息来获取当前同步进度。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:105]:
```suggestion
    * **方法二**：查询下游 TiDB 的 Syncpoint 信息
```

File: ticdc/ticdc-faq.md [hunk 15]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:106]:
```suggestion

        如果下游是 TiDB，并且已启用 [TiCDC Syncpoint 功能](/ticdc/ticdc-upstream-downstream-check.md)，可以通过查询下游 TiDB 的 Syncpoint 信息获取同步进度。

        > **注意：**
        >
        > Syncpoint 信息的更新间隔由 [`sync-point-interval`](/ticdc/ticdc-upstream-downstream-check.md#启用-syncpoint) 参数控制。如需获取最新同步进度，建议使用方法一。

```

File: ticdc/ticdc-faq.md [hunk 16]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+      在下游 TiDB 中执行下面的 SQL 语句：
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:107]:
```suggestion
        在下游 TiDB 中执行以下 SQL 语句，获取上游 TSO (`primary_ts`) 和下游 TSO (`secondary_ts`) 信息。
```

File: ticdc/ticdc-faq.md [hunk 17]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      ```sql
+      select * from tidb_cdc.syncpoint_v1;
+      ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:111]:
```suggestion
        ```sql
        SELECT * FROM tidb_cdc.syncpoint_v1;
        ```
```

File: ticdc/ticdc-faq.md [hunk 18]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      从结果中可以看到上游 TSO (primary_ts) 信息：
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:113]:
```suggestion
        输出结果示例如下：
```

File: ticdc/ticdc-faq.md [hunk 19]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      ```sql
+      +------------------+------------+--------------------+--------------------+---------------------+
+      | ticdc_cluster_id | changefeed | primary_ts         | secondary_ts       | created_at          |
+      +------------------+------------+--------------------+--------------------+---------------------+
+      | default          | syncpoint  | 453879870259200000 | 453879870545461257 | 2024-11-12 20:25:01 |
+      | default          | syncpoint  | 453879948902400000 | 453879949214351361 | 2024-11-12 20:30:01 |
+      | default          | syncpoint  | 453880027545600000 | 453880027751907329 | 2024-11-12 20:35:00 |
+      +------------------+------------+--------------------+--------------------+---------------------+
+      ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:123]:
```suggestion
        ```sql
        +------------------+------------+--------------------+--------------------+---------------------+
        | ticdc_cluster_id | changefeed | primary_ts         | secondary_ts       | created_at          |
        +------------------+------------+--------------------+--------------------+---------------------+
        | default          | syncpoint  | 453879870259200000 | 453879870545461257 | 2024-11-12 20:25:01 |
        | default          | syncpoint  | 453879948902400000 | 453879949214351361 | 2024-11-12 20:30:01 |
        | default          | syncpoint  | 453880027545600000 | 453880027751907329 | 2024-11-12 20:35:00 |
        +------------------+------------+--------------------+--------------------+---------------------+
        ```
```

File: ticdc/ticdc-faq.md [hunk 20]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      表格中 primary_ts 和 secondary_ts 表示的是上游 TiDB 在 primary_ts 时刻的 snapshot 与下游 TiDB 在 secondary_ts 时刻的 snapshot 是相同的。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:125]:
```suggestion
        在输出结果中，每一行表示上游 TiDB 在 `primary_ts` 时刻的 snapshot 与下游 TiDB 在 `secondary_ts` 时刻的 snapshot 一致。
```

File: ticdc/ticdc-faq.md [hunk 21]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      将最新的那条 primary_ts 转换成人类易读的时间格式：
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:127]:
```suggestion
        要查看同步进度，可以将最新的 `primary_ts` 转换为可读的时间格式：
```

File: ticdc/ticdc-faq.md [hunk 22]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      即可得到 primary_ts 对应的时间格式：
+
+      ```sql
+      +------------------------------------+
+      | TIDB_PARSE_TSO(453880027545600000) |
+      +------------------------------------+
+      | 2024-11-12 20:35:00                |
+      +------------------------------------+
+      ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:141]:
````suggestion
        ```sql
        SELECT TIDB_PARSE_TSO(453880027545600000);
        ```

        转换结果示例如下：

        ```sql
        +------------------------------------+
        | TIDB_PARSE_TSO(453880027545600000) |
        +------------------------------------+
        | 2024-11-12 20:35:00                |
        +------------------------------------+
        ```
````

File: ticdc/ticdc-faq.md [hunk 23]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      当 primary_ts 所对应的时间超过了步骤 1 中刚获取的上游 TiDB 集群的 TSO 时，则说明 TiCDC 已同步完所有更新。
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:143]:
```suggestion
        如果 `primary_ts` 对应的时间大于或等于步骤 1 中获取的上游 TiDB 集群的 TSO 时间戳，说明 TiCDC 已经将所有更新同步到下游。
```

File: ticdc/ticdc-faq.md [hunk 24]
```diff
@@ -53,6 +53,99 @@ cdc cli changefeed list --server=http://127.0.0.1:8300
...
+
+      > 注意：
+      >
+      > Syncpoint 信息每隔 [`sync-point-interval`](/ticdc/ticdc-upstream-downstream-check.md#启用-syncpoint) 时间才更新一次，想要及时查看当前同步进度建议通过方法一的 Checkpoint 信息来判断。
+
```
Comment (Oreoxmt) [inline ticdc/ticdc-faq.md:148]:
```suggestion
```
Move this note to the beginning of method 2 for better visibility and to help users choose the appropriate method.

Comment (wlwilliamx) [reply ticdc/ticdc-faq.md]: Thanks for your such valuable suggestions!

**Review by Oreoxmt — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**